oauth.go

```diff
@@ -0,0 +1,52 @@
+package stripe
+
+// Type of OAuth scope.
+type ScopeType string
+
+// List of possible values for OAuth scopes.
+const (
+  ScopeTypeReadWrite ScopeType = "read_write"
+  ScopeTypeReadOnly  ScopeType = "read_only"
+)
+
+// Type of token. This will always be "bearer."
+type OAuthTokenType string
+
+// List of possible OAuthTokenType values.
+const (
+  OAuthTokenTypeBearer  OAuthTokenType = "bearer"
+)
+
+// Params for creating OAuth AuthorizeURL's.
+type AuthorizeURLParams struct {
+  State           *string
+  ResponseType    *string
+  Scope           *string
+  RedirectURI     *string
+  StripeLanding   *string
+  AlwaysPrompt    *bool
+  Express         *bool
+}
+
+// OAuthTokenParams is the set of paramaters that can be used to request
+// OAuthTokens.
+type OAuthTokenParams struct {
+  Params         `form:"*"`
+  GrantType      *string  `form:"grant_type"`
+  Code           *string  `form:"code"`
+  RefreshToken   *string  `form:"refresh_token"`
+  Scope          *string  `form:"scope"`
+  ClientSecret   *string  `form:"client_secret"`
+}
+
+// OAuthToken is the value of the OAuthToken from OAuth flow.
+// https://stripe.com/docs/connect/oauth-reference#post-token
+type OAuthToken struct {
+  AccessToken            string          `json:"access_token"`
```

Comment:
I didn't see which of these were actually deprecated in the docs. Are they officially deprecated, yet?

Reply:
Just wanted to bump this. I was looking here: https://stripe.com/docs/connect/oauth-reference#post-token-response and didn't see any of these deprecated. Thinking we should hide the `refresh_token`, `access_token`, `stripe_publishable_key` ? or mark as deprecated?

Reply:
`refresh_token` and `access_token` are part of the OAuth specification so it would be awkward to not have them at all, but we can mark them as "deprecated" in favor of `stripe_user_id`.

We should probably be consistent with the .NET implementation: https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Entities/OAuth/OAuthToken.cs